service/src/test/java/util/PaymentRequestHelperUtil.java

```diff
@@ -0,0 +1,255 @@
+        }
+        return PSEUDO_CARD_PAN;
+    }
+
+    /**
+     * Request a new pseudocardpan for {@code cardPan} Visa number.
+     * <p>
+     * <b>Note:</b> Pseudocardpan is a unique number for every merchant ID, thus ensure you tested service and these
+     * integration tests use the same payone merchant credentials.
+     * Verify your {@code PAYONE_MERCHANT_ID} and {@code TEST_DATA_PAYONE_MERCHANT_ID} for both applications if
+     * you get "pseudocardpan is not found" response from Payone.
+     *
+     * @param cardPan Visa card number (expected to be test Visa number from Payone)
+     * @param mid Payone Merchant Id
+     * @param aid Payone Sub-Account Id
+     * @param pid Payone Portal Id
+     * @param key Payone Key (Access Token)
+     * @return pseudocardpan string, registered in Payone merchant center
+     * @throws RuntimeException if the response from Payone can't be parsed
+     */
+    public static String fetchPseudoCardPan(String cardPan, String mid, String aid, String pid, String key) {
+        //curl --data "request=3dscheck&mid=$PAYONE_MERCHANT_ID&aid=$PAYONE_SUBACC_ID&portalid=$PAYONE_PORTAL_ID&key=$(md5 -qs $PAYONE_KEY)&mode=test&api_version=3.9&amount=2&currency=EUR&clearingtype=cc&exiturl=http://www.example.com&storecarddata=yes&cardexpiredate=2512&cardcvc2=123&cardtype=V&cardpan=<VISA_CREDIT_CARD_3DS_NUMBER>"
+
+        String cardPanResponse = null;
+        try {
+            cardPanResponse = PayonePostServiceImpl.executePostRequestToString("https://api.pay1.de/post-gateway/",
+                ImmutableList.of(
+                    nameValue("request", "creditcardcheck"),
+                    nameValue("mid", mid),
+                    nameValue("aid", aid),
+                    nameValue("portalid", pid),
+                    nameValue("key", PayoneHash.calculate(key)),
+                    nameValue("mode", "test"),
+                    nameValue("api_version", "3.9"),
+                    nameValue("amount", "40"),
+                    nameValue("currency", "EUR"),
+                    nameValue("clearingtype", "cc"),
+                    nameValue("exiturl", "http://www.example.com"),
+                    nameValue("storecarddata", "yes"),
+                    nameValue("cardexpiredate", "2512"),
+                    nameValue("cardcvc2", "123"),
+                    nameValue("cardtype", "V"),
+                    nameValue("cardpan", cardPan)));
+        } catch (Throwable e) {
+            throw new RuntimeException("Error on pseudocardpan fetch", e);
+        }
+
+        Pattern p = Pattern.compile("^.*pseudocardpan\\s*=\\s*(\\d+).*$", CASE_INSENSITIVE | DOTALL);
+        Matcher m = p.matcher(cardPanResponse);
+
+        if (!m.matches()) {
+            throw new RuntimeException(format("Unexpected pseudocardpan response: %s", cardPanResponse));
+        }
+
+        return m.group(1);
+    }
+
+    public static void cleanupData() {
```

Comment:
We discussed yesterday that the better approach is not to clean up data, but make the tests in a way that they will run no matter what is in the project. Can you remove the clean up so we test this approach with this project?

Reply:
https://github.com/commercetools/commercetools-payone-integration/pull/392/commits/b39674f48631147fc0916ee71ec7bb496d960ff4